sdk/resourcemanager/azure-resourcemanager-appplatform/src/main/java/com/azure/resourcemanager/appplatform/implementation/SpringAppDeploymentImpl.java

```diff
@@ -324,7 +367,7 @@ public SpringAppDeploymentImpl withoutEnvironment(String key) {
 
     @Override
     public SpringAppDeploymentImpl withVersionName(String versionName) {
-        ensureSource();
+        ensureSource(UserSourceType.fromString("Other"));
```

Comment:
Many just a null? To avoid service someday decide to add an `Other` as enum.

Reply:
Changed.